tools/fastq_filtration.py

```diff
@@ -0,0 +1,117 @@
+    - quality_threshold (int): allow filter sequence and it is the bottom bound. Has got default value: 0. 
+
+    Return:
+    - str, if fastq-seq corresponds filtration condition.
+    - None in opposite situation.
+    """
+    seq = seq_and_quality[0]
+    quality_seq = seq_and_quality[1]
+
+    q_score = []
+    for symbol in quality_seq:
+        q_score.append(ord(symbol)-33)
+
+    quality = round(sum(q_score)/len(q_score), 4)
+    if quality >= quality_threshold:
+        return seq
+
+
+def filtre_by_length(seq:str, length_bounds:tuple) -> str:
+    """
+    Filter sequence by length.
+
+    Arguments:
+    - seq (str): fastq-seq
+    - length_bounds (tuple): bottom and top bound to filtration. Has got default value: (0, 2**32).
+
+    Return:
+    - str, if fastq-seq corresponds filtration condition.
+    - None in opposite situation.
+    """
+    length_seq = len(seq)
+    if length_seq >= length_bounds[0] and length_seq <= length_bounds[1]:
+        return seq
+
+
+
+def filter_by_gc_level(seq:str, gc_bounds:tuple) -> str:
+    """
+    Filter sequence by gc-level.
+
+    Arguments:
+    - seq (str): fastq-seq
+    - gc_bounds (tuple): bottom and top bound to filtration. Use only top bound if gc_bound is int.
+
+    Return:
+    - str, if fastq-seq corresponds filtration condition.
+    - None in opposite situation. 
+    """
+    if type(gc_bounds) == int:
+        bottom_bound = 0
+        top_bound = gc_bounds
+    elif type(gc_bounds) == tuple:
+        bottom_bound = gc_bounds[0]
+        top_bound = gc_bounds[1]
+
+    gc_amount = 0
+    for nucl in seq:
+        if nucl == "G" or nucl == "C":
+            gc_amount += 1
+    
+    cur_gc_level = round(gc_amount / len(seq) * 100, 4)
+    if cur_gc_level >= bottom_bound and cur_gc_level <= top_bound:
+        return seq
+
+
+    
+
+def filter_fastq(dict_of_fastq:dict, gc_bounds:tuple, length_bounds=(0, 2**32), quality_threshold=0) -> list:
+    """
+    Function containing conditions to filtration fastq-sequences.
+    
+    Takes arbitrary number of arguments with characteristic to filtrarion.
+    Returns the result of the filtration in dictionary.
+
+    Arguments:
+    - dict_of_fastq (dict): dictionary with fastq-seq in format 'name' : ('sequence', 'quality')
+    - gc_bounds (tuple or int): bottom and top bound to filtration. Use only top bound if gc_bound is int
+    - length_bounds (tuple): bottom and top bound to filtration. Has got default value: (0, 2**32).
+    - quality_threshold (int): allow filter sequence and it is the bottom bound. Has got default value: 0.
+    """
+
+    seqs_with_qualities = [pair for pair in dict_of_fastq.values()]
+
+    pairs = []
+    for pair in seqs_with_qualities:
+        if (filter_by_gc_level(pair[0], gc_bounds) and filtre_by_length(pair[0], length_bounds) and filter_by_quality(pair, quality_threshold)) != None:
+            pairs.append(pair)
+
+    ans_dict = {}
```

Comment:
```suggestion
    results = {}
```
1) В программировании все таки не ответ а результат
2) Множественное число
3) Не очень принято добавлять к переменной ее тип данных в названии, кроме каких-то отдельных случаев. Мы же не пишем `seq_str` и `num_int`.